### ISSUE TITLE
Many-to-Many LSTM predicting same output for each timestep

### ISSUE THREAD

**Issue by Adizx12** [keras-team/keras#8493]
Hello,
I'm trying to prepare many-to-many LSTM model, where for each timestep model should predict separate output. I've tried to use Sequential model, however faced a strange problem.

Model definition:
```python
        model = Sequential()
        model.add(Masking(mask_value=-1, input_shape=(timesteps, input_dimension)))
        model.add(LSTM(hidden_neurons_number, return_sequences=True))
        for layer in range(0, self.hidden_layers-1):
            model.add(LSTM(hidden_neurons_number, return_sequences=True'))
        model.add(Dropout(0.2))
        model.add(TimeDistributed(Dense(output_dimension)))
        # model.add(LSTM(output_dimension, return_sequences=True))
        model.add(Activation("sigmoid"))    # normalize

        adam = optimizers.Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=1e-08, decay=0.0)
        # model.compile(loss="binary_crossentropy", optimizer="adam")
        model.compile(loss="mse", optimizer="RMSprop")


        self.model.fit(model_input, expected_output, batch_size=1, epochs=epochs, callbacks=model_callbacks)

        generated_performance = self.model.predict(model_input, batch_size=1)
```

Sample input:
![obraz](https://user-images.githubusercontent.com/19515958/32816171-bc2dd530-c9b7-11e7-8ff0-bd360385ec65.png)
(First is number marking time position, then binary encoding of events (1 - present, 0 - not)

Expected output: (passed for learning)
![obraz](https://user-images.githubusercontent.com/19515958/32816271-28998034-c9b8-11e7-9902-d0c98fb43ff2.png)
(For each timestep output should be array of normalized values)

Learning progression:
![obraz](https://user-images.githubusercontent.com/19515958/32816345-730af72e-c9b8-11e7-98bc-83d7e2921856.png)
![obraz](https://user-images.githubusercontent.com/19515958/32816356-7f3381d8-c9b8-11e7-9e9f-ac07ac9650a6.png)

And finally generated output:
![obraz](https://user-images.githubusercontent.com/19515958/32816375-a81d2d2e-c9b8-11e7-999f-7342b0a7afe0.png)
(Each next timestep has identical values)

So the question is: Why only the first few timesteps have different values, then all further ones are exactly the same? Why mse is stabilized when output is completely different?

Testing sequence has 200 timesteps, for evaluating purposes same data is passed in single batch for a few generations to see if model can replicate single data file.

If more info would be needed please write it out, I'm not sure what more is important for the case.

Thanks in advance for any help!
